smart-contracts/assembly/contracts/FT/__tests__/ft-burn.spec.ts

```diff
@@ -92,3 +109,61 @@ describe('Fails burn ERC20', () => {
     burn(new Args().add(u256.Max).serialize()),
   );
 });
+
+const allowAmount = new u256(1, 1, 1, 1);
+describe('burnFrom', () => {
+  beforeAll(() => {
+    switchUser(user3Address);
+
+    // Increase allowance for U1 to spend U3 tokens
+    increaseAllowance(
+      new Args().add(user1Address).add(allowAmount).serialize(),
+    );
+    switchUser(user1Address);
+  });
+
+  throws('Fails because not enough allowance U3 => U1 ', () => {
+    burnFrom(
+      new Args()
+        .add(user3Address)
+        .add(user2Address)
+        // @ts-ignore
+        .add(allowAmount + u256.One)
+        .serialize(),
+    );
+  });
+
+  throws('Fails because not enough token on U3', () =>
+    burnFrom(
+      new Args()
+        .add(user3Address)
+        .add(user2Address)
+        .add(allowAmount)
+        .serialize(),
+    ),
+  );
+
+  test('u1 sends tokens to u3 then burnFrom all tokens', () => {
```

Comment:
Can you clarify what you are testing from a higher point of view? Here I have the feeling that you are explaining what the test does.